src/Stratis.Bitcoin/Utilities/Extensions/DBreezeExtensions.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Threading.Tasks;
+using DBreeze;
+using DBreeze.Exceptions;
+using DBreeze.Transactions;
+using Microsoft.Extensions.Logging;
+
+namespace Stratis.Bitcoin.Utilities.Extensions
+{
+    public class RetryOptions
+    {
+        public RetryOptions()
+        {
+            this.RetryCount = 1;
+            this.Delay = TimeSpan.FromMilliseconds(100);
+        }
+
+        public RetryOptions(short retryCount, TimeSpan delay)
+        {
+            if (retryCount < 1)
+                throw new ArgumentOutOfRangeException(nameof(retryCount), "Retry count cannot be less or equal to 0");
+
+            this.RetryCount = retryCount;
+            this.Delay = delay;
+        }
+
+        public static RetryOptions Default => new RetryOptions();
+
+        public short RetryCount { get; set; }
+
+        public TimeSpan Delay { get; }
+
+        public override string ToString()
+        {
+            return $"Retry count: {this.RetryCount}; Delay: {this.Delay}";
+        }
+    }
+
+    public static class DBreezeExtensions
+    {
+        public static void TransactionalCommit(this DBreezeEngine dbreeze, ILogger logger, Action<Transaction> handler, RetryOptions retryOptions = null)
+        {
+            if (retryOptions == null || retryOptions.RetryCount == 0)
+            {
+                ExecuteTransaction(dbreeze, handler);
+                return;
+            }
+
+            while (retryOptions.RetryCount > 0)
+            {
+                try
+                {
+                    retryOptions.RetryCount--;
+                    ExecuteTransaction(dbreeze, handler);
+                    return;
+                }
+                catch (TableNotOperableException ex)
+                {
+                    logger.LogError("Failed to commit transaction. Retrying.", ex);
+                    Task.Delay(retryOptions.Delay).GetAwaiter().GetResult();
+                    ExecuteTransaction(dbreeze, handler);
+                }
+            }
+        }
+
+        private static void ExecuteTransaction(DBreezeEngine dbreeze, Action<Transaction> handler)
```

Comment:
@bokobza @noescape00 @fassadlr @MithrilMan 
I suggest the retry logic should not create the `dbreeze.GetTransaction()` but let it happen at the point where the code is calling the db effectively making this component only responsible for retires.

Thoughts?